### PR TITLE
improve debug logs for the `mesheryctl pattern apply` cmd

### PR DESCRIPTION
**Pull request by ayushthe1**
Signed-off-by: ayush_gitk <ayushsharmaa101@gmail.com>

**Notes for Reviewers**

This PR intends to fix the debug logs for the `mesheryctl pattern apply` cmd.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Thanks for this, @ayushthe1. There are a few lint checks to investigate.

**Comment by ayushthe1** [comment]
> Thanks for this, @ayushthe1. There are a few lint checks to investigate.

i will soon fix that.  Was actually busy in the build meet ,so forgot about it.

**Comment by Chadha93** [comment]
@ayushthe1  Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Comment by ayushthe1** [comment]
@alphaX86 ,could you please review it and suggest any changes.

**Comment by acald-creator** [comment]
Rebase this PR to pull in the latest changes from the meshmodel state.

Hopefully this will satisfy the `golangci-lint` complaint.

**Comment by ayushthe1** [comment]
@acald-creator I updated the pr by merging the master into it but the error is still coming.

**Comment by acald-creator** [comment]
@ayushthe1, I believe @Revolyssup can provide some insight on this as he's been working on the MeshModal in `server/*`

File: mesheryctl/internal/cli/root/pattern/apply.go [hunk 1]
```diff
@@ -66,12 +66,12 @@ mesheryctl pattern apply [pattern-name]
...
 			req, err = utils.NewRequest("GET", patternURL+"?search="+patternName, nil)
 			if err != nil {
-				return err
+				return errors.Wrap(err, "could not create request ")
 			}
 
 			resp, err := client.Do(req)
 			if err != nil {
-				return err
+				return errors.Errorf("unable to reach Meshery server at %s. Verify your environment's readiness for a Meshery deployment by running `mesheryctl system check`", mctlCfg.GetBaseMesheryURL())
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/pattern/apply.go:74]:
```suggestion
				return errors.Errorf("unable to reach Meshery Server at %s. Verify your environment's readiness for a Meshery deployment by running `mesheryctl system check`", mctlCfg.GetBaseMesheryURL())
```

File: mesheryctl/internal/cli/root/pattern/apply.go [hunk 2]
```diff
@@ -104,7 +104,7 @@ mesheryctl pattern apply [pattern-name]
 			if validURL := govalidator.IsURL(file); !validURL {
 				content, err := os.ReadFile(file)
 				if err != nil {
-					return err
+					return errors.Errorf("provided file path %s is invalid. enter a valid path ", file)
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/pattern/apply.go:107]:
```suggestion
					return errors.Errorf("file path %s is invalid. Enter a valid path ", file)
```

File: mesheryctl/internal/cli/root/pattern/apply.go [hunk 3]
```diff
@@ -189,12 +189,12 @@ mesheryctl pattern apply [pattern-name]
...
 				req, err = utils.NewRequest("POST", patternURL, bytes.NewBuffer(jsonValues))
 				if err != nil {
-					return err
+					return errors.Wrap(err, "could not create request ")
 				}
 
 				resp, err := client.Do(req)
 				if err != nil {
-					return err
+					return errors.Errorf("unable to reach Meshery server at %s. Verify your environment's readiness for a Meshery deployment by running `mesheryctl system check`", mctlCfg.GetBaseMesheryURL())
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/pattern/apply.go:197]:
```suggestion
					return errors.Errorf("unable to reach Meshery Server at %s. Verify your environment's readiness for a Meshery deployment by running `mesheryctl system check`", mctlCfg.GetBaseMesheryURL())
```

File: mesheryctl/internal/cli/root/pattern/apply.go [hunk 4]
```diff
@@ -232,7 +232,7 @@ mesheryctl pattern apply [pattern-name]
 		s.Start()
 		res, err := client.Do(req)
 		if err != nil {
-			return err
+			return errors.Errorf("unable to reach Meshery server at %s. Verify your environment's readiness for a Meshery deployment by running `mesheryctl system check`", mctlCfg.GetBaseMesheryURL())
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/pattern/apply.go:235]:
```suggestion
			return errors.Errorf("unable to reach Meshery Server at %s. Verify your environment's readiness for a Meshery deployment by running `mesheryctl system check`", mctlCfg.GetBaseMesheryURL())
```

**Review by leecalcote — COMMENTED** [review]
Always capitalize Meshery component names, like Meshery Server.

**Review by leecalcote — APPROVED** [review]
Thanks, @ayushthe1

**Pull request merged**